lib/src/geo_collection_reference.dart

```diff
@@ -282,15 +288,28 @@ class GeoCollectionReference<T> {
     required final String field,
     required final String geohashField,
     final Query<T>? Function(Query<T> query)? queryBuilder,
+    final bool isCacheFirst = false,
   }) {
     return _geohashes(radiusInKm: radiusInKm, center: center).map(
       (final geohash) async {
-        final querySnapshot = await geoQuery(
+        late QuerySnapshot<T> querySnapshot;
+        final query = geoQuery(
           field: field,
           geohashField: geohashField,
           geohash: geohash,
           queryBuilder: queryBuilder,
-        ).get();
+        );
+        try {
+          querySnapshot = await query.get(
+            GetOptions(
+              source: isCacheFirst ? Source.cache : Source.serverAndCache,
+            ),
+          );
+        } on FirebaseException catch (_) {
+          if (isCacheFirst) {
+            querySnapshot = await query.get();
+          }
```

Comment:
When `isCacheFirst` is `false`, the `FirebaseException` caught here is ignored, so how about `rethrow` it in `else` phrase?

```suggestion
          if (isCacheFirst) {
            querySnapshot = await query.get();
          } else {
            rethrow;
          }
```

Reply:
Hi, great remark, I haven't noticed that, thanks!
Can you have a look at the refactored version from my latest commit? I used your suggestion and refactored `catch` section a bit so it's more concise now.

Reply:
@Kurczaak 

Cool! CI unit test also passes, so I will approve now!